zerver/webhooks/gitlab/view.py

```diff
@@ -33,7 +33,7 @@ def get_push_event_body(payload: Dict[str, Any]) -> str:
 
 def get_normal_push_event_body(payload: Dict[str, Any]) -> str:
     compare_url = '{}/compare/{}...{}'.format(
-        get_repository_homepage(payload),
+        get_project_homepage(payload),
```

Comment:
Can you explain the reasoning for why this is better in your commit message?  Or do they always have the same value?  Certainly there are no test changes in this commit, which makes me think they might be the same?

Reply:
AFAICT all webhooks have a `project` object, but not all of them have a `repository` object. All the values I've witnessed have been the same.
Without this change clicking the test button in Gitlab fails for system hooks.